helidon-linker/src/main/java/io/helidon/linker/StartScript.java

```diff
@@ -0,0 +1,621 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.linker;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+import io.helidon.linker.util.FileUtils;
+import io.helidon.linker.util.Log;
+import io.helidon.linker.util.ProcessMonitor;
+import io.helidon.linker.util.StreamUtils;
+
+import static io.helidon.linker.util.Constants.CDS_REQUIRES_UNLOCK_OPTION;
+import static io.helidon.linker.util.Constants.CDS_SUPPORTS_IMAGE_COPY;
+import static io.helidon.linker.util.Constants.CDS_UNLOCK_OPTIONS;
+import static io.helidon.linker.util.Constants.DIR_SEP;
+import static io.helidon.linker.util.Constants.EOL;
+import static io.helidon.linker.util.Constants.OSType.Windows;
+import static io.helidon.linker.util.Constants.OS_TYPE;
+import static io.helidon.linker.util.FileUtils.assertDir;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Installs a start script for a main jar.
+ */
+public class StartScript {
+    private static final String INSTALL_PATH = "start";
+    private final Path installDirectory;
+    private final Path scriptFile;
+    private final String script;
+
+    /**
+     * Returns a new builder.
+     *
+     * @return The builder.
+     */
+    static Builder builder() {
+        return new Builder();
+    }
+
+    private StartScript(Builder builder) {
+        this.installDirectory = builder.scriptInstallDirectory;
+        this.scriptFile = builder.scriptFile;
+        this.script = builder.script;
+    }
+
+    /**
+     * Returns the install directory.
+     *
+     * @return The directory.
+     */
+    Path installDirectory() {
+        return installDirectory;
+    }
+
+    /**
+     * Install the script.
+     *
+     * @return The path to the installed script.
+     */
+    Path install() {
+        try {
+            Files.copy(new ByteArrayInputStream(script.getBytes(StandardCharsets.UTF_8)), scriptFile);
```

Comment:
Shouldn't be the next?:
try(InputStream input = new ByteArrayInputStream(script.getBytes(StandardCharsets.UTF_8))){
    Files.copy(input, scriptFile);

Reply:
Normally yes, but... `ByteArray` streams don't need to be closed.